src/components/Filters/StatefulFilters.tsx

```diff
@@ -73,9 +76,17 @@ export class StatefulFilters extends React.Component<StatefulFiltersProps, State
       }
     });
 
-    Promise.all(promises).then(types => {
-      this.setState({ filterTypes: types });
-    });
+    this.filterTypePromise = makeCancelablePromise(Promise.all(promises));
+    this.filterTypePromise.promise
+      .then(types => {
+        this.setState({ filterTypes: types });
+        this.filterTypePromise = undefined;
+      })
+      .catch(err => {
+        if (!err.isCancelable) {
+          console.debug(err);
+        }
```

Comment:
Not sure, but shouldn't `err` be thrown in order to be propagated to the other `catch` blocks? Or is it not necessary?